frontend/templates/product-list/hooks/useCurrentProductList.tsx

```diff
@@ -0,0 +1,75 @@
+import { ProductList } from '@models/product-list';
+import { useVariantProductList } from './useVariantProductList';
+import { useItemTypeProductList } from './useItemTypeProductList';
+import React from 'react';
+import { SentryError } from '@ifixit/sentry';
+
+/**
+ * Used to modify product list data before rendering
+ * The Order that the hooks are called is important as to what the data will look like
+ * This lets us do things like Item Type Overrides for product lists
+ */
+// export function useCurrentProductList(
+//    productList: ProductList,
+//    algoliaUrl?: string
+// ): ProductList {
+//    const variantUpdatedProductList = useVariantProductList(
+//       productList,
+//       algoliaUrl
+//    );
+//    const itemTypeUpdatedProductList = useItemTypeProductList(
+//       variantUpdatedProductList
+//    );
+//    return itemTypeUpdatedProductList ?? variantUpdatedProductList;
+// }
+
+type CurrentProductListContext = {
+   currentProductList: ProductList;
+   setCurrentProductList: (_: ProductList) => void;
```

Comment:
setState has more possibilities in the type than this. You can just do:

```suggestion
   setCurrentProductList: Dispatch<SetStateAction<ProductList>>
```

Reply:
It looks like this value is never used.

```suggestion
```